02_snake_animation/main.go

```diff
@@ -39,16 +41,37 @@ func drawSnake(s snake) {
 	termbox.Flush()
 }
 
-// Makes a move for a snake and returns a snake with an updated position.
-func moveSnake(s snake) snake {
+// Makes a move for a snake and returns a snake with an updated position
+// and
+// Makes a single iteration for a snake
+func stepForward(s snake) snake {
+	s.dir = coord{1, 0}
 	s.pos.x += s.dir.x
 	s.pos.y += s.dir.y
+	drawSnake(s)
+	return s
+}
+
+func stepBack(s snake) snake {
+	s.dir = coord{1, 0}
+	s.pos.x -= s.dir.x
+	s.pos.y -= s.dir.y
```

Comment:
1. `s.pos.y -= s.dir.y` is a useless statement, because `s.dir.y` is zero.
2. 

```
s.dir = coord{1, 0}
s.pos.x -= s.dir.x
```

is equivalent to 

```
s.dir = coord{-1, 0}
s.pos.x += s.dir.x
```

If you replace `-=` with `+=` where possible, then the following commands can be moved outside `step*` functions:

```
	s.pos.x += s.dir.x
	s.pos.y += s.dir.y
```

This is a generic logic that is performed at every step and it is not specific to a movement direction.